### PR TITLE
ENH: Conform to BIDS Derivatives as of BIDS 1.4.0

### PR DESCRIPTION
**Pull request by effigies**
## Changes proposed in this pull request

* Check through derivatives for things not addressed by the standard, add to `.bidsignore`
* Update function that generates `dataset_description.json`
* ...

## Documentation that should be reviewed

Should check the "Outputs of fMRIPrep" section if any changes need to be made.

### CODE REVIEW

File: fmriprep/utils/bids.py [hunk 1]
```diff
@@ -29,30 +43,31 @@ def write_derivative_description(bids_dir, deriv_dir):
...
     if fname.exists():
-        with fname.open() as fobj:
-            orig_desc = json.load(fobj)
+        orig_desc = json.loads(fname.read_text())
 
     if 'DatasetDOI' in orig_desc:
-        desc['SourceDatasetsURLs'] = ['https://doi.org/{}'.format(
-            orig_desc['DatasetDOI'])]
+        desc['SourceDatasets'] = [{
+            'URL': 'https://doi.org/{}'.format(orig_desc['DatasetDOI']),
```
Comment (mgxd) [inline fmriprep/utils/bids.py:64]:
```suggestion
            'URL': f'https://doi.org/{orig_desc["DatasetDOI"]}',
```

File: fmriprep/utils/bids.py [hunk 2]
```diff
@@ -29,30 +43,31 @@ def write_derivative_description(bids_dir, deriv_dir):
...
     if 'FMRIPREP_SINGULARITY_URL' in os.environ:
-        singularity_url = os.environ['FMRIPREP_SINGULARITY_URL']
-        desc['SingularityContainerURL'] = singularity_url
-
-        singularity_md5 = _get_shub_version(singularity_url)
-        if singularity_md5 and singularity_md5 is not NotImplemented:
-            desc['SingularityContainerMD5'] = _get_shub_version(singularity_url)
+        desc['GeneratedBy'][0]['Container'] = {
+            "Type": "singularity",
+            "URI": os.environ['FMRIPREP_SINGULARITY_URL']
```
Comment (mgxd) [inline fmriprep/utils/bids.py:53]:
```suggestion
            "URI": os.getenv('FMRIPREP_SINGULARITY_URL')
```

File: fmriprep/utils/bids.py [hunk 3]
```diff
@@ -7,20 +7,34 @@
...
 
+def write_bidsignore(deriv_dir):
+    bids_ignore = [
+        "*.html", "logs/", "figures/",  # Reports
+        "*_xfm.*",  # Unspecified transform files
+        "*.surf.gii",  # Unspecified structural outputs
+        # Unspecified functional outputs
+        "*_boldref.nii.gz", "*_bold.func.gii",
+        "*_mixing.tsv", "*_AROMAnoiseICs.csv", "*_regressors.tsv",
+    ]
```
Comment (mgxd) [inline fmriprep/utils/bids.py:18]:
```suggestion
def write_bidsignore(deriv_dir):
    bids_ignore = (
        "*.html", "logs/", "figures/",  # Reports
        "*_xfm.*",  # Unspecified transform files
        "*.surf.gii",  # Unspecified structural outputs
        # Unspecified functional outputs
        "*_boldref.nii.gz", "*_bold.func.gii",
        "*_mixing.tsv", "*_AROMAnoiseICs.csv", "*_regressors.tsv",
    )
```

File: fmriprep/cli/run.py [hunk 4]
```diff
@@ -166,6 +166,7 @@ def main():
         write_derivative_description(
             config.execution.bids_dir, config.execution.output_dir / "fmriprep"
         )
+        write_bidsignore(config.execution.output_dir / "fmriprep")
```
Comment (oesteban) [inline fmriprep/cli/run.py:169]: should we add another `.bidsignore` at the top level containing `freesurfer/`? Or at least when `--fs-subjects-dir` was not specified?

Comment (effigies) [reply fmriprep/cli/run.py:169]: The directory containing FreeSurfer is not a BIDS dataset.

**Comment by effigies** [comment]
Requires nipreps/niworkflows#560.

**Comment by effigies** [comment]
@mgxd @oesteban I think this is ready for review.

As previously discussed, we'll aim for getting filenames and REQUIRED metadata straight. Any RECOMMENDED or OPTIONAL metadata can be a future enhancement.

File: setup.cfg [hunk 5]
```diff
@@ -27,7 +27,7 @@ install_requires =
     nipype >= 1.5.1
     nitime
     nitransforms >= 20.0.0rc3,<20.2
-    niworkflows @ git+https://github.com/nipreps/niworkflows.git@master
+    niworkflows @ git+https://github.com/effigies/niworkflows.git@bids/timeseries
```
Comment (effigies) [inline setup.cfg:30]:
```suggestion
    niworkflows @ git+https://github.com/nipreps/niworkflows.git@master
```

**Comment by effigies** [comment]
Yup. I'll point to derivatives + BEPs 011 and 012. Thanks.

**Comment by effigies** [comment]
Docs look good. I'll go ahead and merge to keep things moving...

**Pull request merged**